src/OrchardCore/OrchardCore.Modules.Abstractions/Manifest/ManifestAttributes.cs

```diff
@@ -0,0 +1,114 @@
+using System;
+using System.Collections.Generic;
+
+namespace OrchardCore.Modules.Manifest
+{
+    /// <summary>
+    /// Defines a Module which is composed of features.
+    /// If the Module has only one default feature, it may be defined there.
+    /// </summary>
+
+    [AttributeUsage(AttributeTargets.Assembly, AllowMultiple = false, Inherited = false)]
+    public class ModuleAttribute : Attribute
+    {
+        /// <param name="Name">
+        /// Human-readable name of the module. If not provided, the assembly name will be used.
+        /// </param>
+        /// <param name="Author">The developer of the module.</param>
+        /// <param name="Website">The URL for the website of the module developer.</param>
+        /// <param name="Version">The version number of the module in SemVer format.</param>
+        /// <param name="Tags">A comma-separated lists of tags for the module.</param>
+        /// <param name="Description">A brief summary of what the default feature does.</param>
+        /// <param name="Dependencies">
+        /// A comma-separated list of features that the default feature depends on.
+        /// So that its drivers / handlers are invoked after those of dependencies.
+        /// </param>
+        /// <param name="Priority">
+        /// The priority of the default feature without breaking the dependencies order.
+        /// higher is the priority, later the drivers / handlers are invoked.
+        /// </param>
+        /// <param name="Category">
+        /// The group (by category) that the default feature belongs.
+        /// If not provided, defaults to 'Uncategorized'.
+        /// </param>
+
+        public ModuleAttribute(
+            string Name = null,
```

Comment:
Since all these arguments are optional, would it make sense to remove the constructor and use settable properties? (it would also solve the inconsistent casing :sweat_smile:)

Reply:
This was my initial though when I saw the lower case usage, I thought they were properties.

Reply:
😄 

- But which syntax do we prefer? No problem i can change it.


      [assembly: Module(
          Name: "Body",
          Author: "The Orchard Team",
          ...
      )]

      [assembly: Module(
          Name = "Body",
          Author = "The Orchard Team",
          ...
      )]

- Then, for the properties it allows to use `{ get; }` in place of  `{ get; set; }`

- Finally, because a module can define a default feature, in the module ctor i create a new inner feature by using some of the module ctor params. First i derivated module from feature but in the following code, the features was including the module itself. But i can easily filter it.

      var module = Assembly.GetCustomAttribute<ModuleAttribute>();
      var features = Assembly.GetCustomAttributes<FeatureAttribute>();

**So, let me know which syntax you prefer.**

Reply:
Yes, i was thinking the same like this `new ModuleAttribute() { Name = ... }`. But it seems that now it allows the above. I will retry

Reply:
Yes, it is allowed, at least it builds. So which syntax do you prefer?

- Anyway i first made the module attribute inheriting from the feature attribute.

- Then here a very "little" problem, i don't want to propose the `Id` when defining a module because we set it at runtime with the assembly name. So, i would want to make this property internal but still allowing to define it for a feature. I can do that but with a ctor parameter.

- So, if we use settable props, maybe still use a ctor param for the feature `Id`, OR for a module give an intellisense comment saying that it can be skipped because it is set at runtime.